backend/de.metas.async/src/main/java/de/metas/async/api/IAsyncBatchBL.java

```diff
@@ -110,7 +113,17 @@
 			@NonNull List<T> model,
 			@NonNull String asyncBatchInternalName);
 
+	IAutoCloseable assignTempAsyncBatchIdToModel(@NonNull Object model, @NonNull AsyncBatchId asyncBatchId);
+
 	I_C_Async_Batch getAsyncBatchById(AsyncBatchId asyncBatchId);
 
 	AsyncBatchId newAsyncBatch(String asyncBatchType);
+
+	Optional<String> getAsyncBatchTypeInternalName(@NonNull final I_C_Async_Batch asyncBatch);
+
+	boolean isAsyncBatchTypeInternalName(@NonNull I_C_Async_Batch asyncBatch, @NonNull String expectedInternalName);
+
+	Optional<AsyncBatchType> getAsyncBatchType(@NonNull I_C_Async_Batch asyncBatch);
+
+	AsyncBatchType getAsyncBatchTypeById(@NonNull AsyncBatchTypeId asyncBatchTypeId);
```

Comment:
can be done later: this should be in `AsyncBatchTypeRepository`

Reply:
basically, 

* getAsyncBatchTypeById shall go to a new AsyncBatchTypeRepository.
* getAsyncBatchType shall stay here.